# HG changeset patch
# User Rik <rik@octave.org>
# Date 1497456953 25200
#      Wed Jun 14 09:15:53 2017 -0700
# Node ID 0f4d3b06464cca461118fcbb532101364429f8ca
# Parent  b63c3a09aee7aa8791131fdfabb3d936983c8b42
MatrixType.h: Deprecate is_diagonal, is_upper_triangular, is_lower_triangular, is_banded.

* MatrixType.h: Deprecate is_diagonal, is_upper_triangular,
is_lower_triangular, is_banded and replace with Octave names isdiag, istriu,
istril, isbanded.

diff --git a/liboctave/array/MatrixType.h b/liboctave/array/MatrixType.h
--- a/liboctave/array/MatrixType.h
+++ b/liboctave/array/MatrixType.h
@@ -102,28 +102,44 @@ public:
   double band_density (void) const { return bandden; }
 
   int nupper (void) const { return upper_band; }
 
   int nlower (void) const { return lower_band; }
 
   bool is_dense (void) const { return dense; }
 
-  bool is_diagonal (void) const
+  bool isdiag (void) const
   { return (typ == Diagonal || typ == Permuted_Diagonal); }
 
-  bool is_upper_triangular (void) const
+  OCTAVE_DEPRECATED ("use 'isdiag' instead")
+  bool is_diagonal (void) const
+  { return isdiag (); }
+
+  bool istriu (void) const
   { return (typ == Upper || typ == Permuted_Upper); }
 
-  bool is_lower_triangular (void) const
+  OCTAVE_DEPRECATED ("use 'istriu' instead")
+  bool is_upper_triangular (void) const
+  { return istriu (); }
+
+  bool istril (void) const
   { return (typ == Lower || typ == Permuted_Lower); }
 
-  bool is_banded (void)
+  OCTAVE_DEPRECATED ("use 'istril' instead")
+  bool is_lower_triangular (void) const
+  { return istril (); }
+
+  bool isbanded (void) const
   { return (typ == Banded || typ == Banded_Hermitian); }
 
+  OCTAVE_DEPRECATED ("use 'isbanded' instead")
+  bool is_banded (void) const
+  { return isbanded (); }
+
   bool is_tridiagonal (void) const
   { return (typ == Tridiagonal || typ == Tridiagonal_Hermitian); }
 
   bool ishermitian (void) const
   {
     return (typ == Banded_Hermitian || typ == Tridiagonal_Hermitian
             || typ == Hermitian);
   }
